dist/joii.js

```diff
@@ -922,18 +1050,223 @@ JOII.CreateProperty = function(obj, name, val, writable) {
     }
 };
 
+
+
+/**
+ * Adds a function to the prototype. 
+ *
+ * Meta.parameters should include the types in order to support overloading.
+ *
+ * @param {Object}   prototype
+ * @param {Object}   meta
+ * @param {Function} fn
+ * @param {Boolean}  ignoreDuplicate
+ */
+JOII.AddFunctionToPrototype = function(prototype, meta, fn, ignoreDuplicate) {
+
+    if (typeof (ignoreDuplicate) === 'undefined') {
+        ignoreDuplicate = false;
+    }
+    if (typeof (prototype.__joii__.metadata[meta.name]) !== 'object') {
+        prototype.__joii__.metadata[meta.name] = JOII.Compat.extend(true, {}, meta);
+        prototype.__joii__.metadata[meta.name].overloads = [];
+    }
+
+    var proto_meta = prototype.__joii__.metadata[meta.name];
+
+    if (typeof (proto_meta.overloads) !== 'object') {
+        proto_meta.overloads = [];
+    }
+
+    if (proto_meta.visibility !== meta.visibility) {
+        throw 'Member ' + meta.name + ': inconsistent visibility.';
+    }
+
+    proto_meta.is_abstract = false;
+
+    for (var i = 0; i < meta.parameters.length - 1; i++) {
+        if (meta.parameters[i] == '...') {
+            throw 'Member ' + meta.name + ': Variadic parameter (...) must be the last in the function parameter list.';
+        }
+    }
+
+    for (var idx = 0; idx < proto_meta.overloads.length; idx++) {
+        var functionParametersMeta = proto_meta.overloads[idx];
```

Comment:
Please be consistent with codestyle in variable names.

- Variables are written in snake_case
- Classes (functions that can be instantiated using the `new` keyword)  are written in PascalCase (UpperCamelCase)
- Methods are written in camelCase.

The reason for this convention is so you can always see what a variable is without having to look it up.
This also applies to the rest of the commit.


I know a lot of the older code also doesn't fully comply to these rules (yet), but as the codebase grows, I think it's a good idea to comply to this sooner than later.